projects/gateway/pkg/services/k8sadmission/validating_admission_webhook.go

```diff
@@ -395,160 +394,75 @@ func (wh *gatewayValidationWebhook) validateAdmissionRequest(
 	isDelete := admissionRequest.Operation == v1beta1.Delete
 	dryRun := isDryRun(admissionRequest)
 
-	switch gvk {
-	case ListGVK:
+	if gvk == ListGVK {
 		return wh.validateList(ctx, admissionRequest.Object.Raw, dryRun)
-	case gwv1.GatewayGVK:
-		if isDelete {
-			// we don't validate gateway deletion
-			break
-		}
-		return wh.validateGateway(ctx, admissionRequest)
-	case gwv1.VirtualServiceGVK:
-		if isDelete {
-			err := wh.validator.ValidateDeleteVirtualService(ctx, ref, dryRun)
-			if err != nil {
-				return &validation.Reports{}, &multierror.Error{Errors: []error{err}}
-			}
-		} else {
-			return wh.validateVirtualService(ctx, admissionRequest)
-		}
-	case gwv1.RouteTableGVK:
-		if isDelete {
-			err := wh.validator.ValidateDeleteRouteTable(ctx, ref, dryRun)
-			if err != nil {
-				return &validation.Reports{}, &multierror.Error{Errors: []error{err}}
-			}
-		} else {
-			return wh.validateRouteTable(ctx, admissionRequest)
-		}
-	case gloov1.UpstreamGVK:
-		if isDelete {
-			err := wh.validator.ValidateDeleteUpstream(ctx, ref, dryRun)
-			if err != nil {
-				return &validation.Reports{}, &multierror.Error{Errors: []error{err}}
-			}
-		} else {
-			return wh.validateUpstream(ctx, admissionRequest)
-		}
-	case gloov1.SecretGVK:
-		// We only support validation of secrets for DELETE operations
-		if isDelete {
-			err := wh.validator.ValidateDeleteSecret(ctx, ref, dryRun)
-			if err != nil {
-				return &validation.Reports{}, &multierror.Error{Errors: []error{err}}
-			}
-		}
 	}
-	return &validation.Reports{}, nil
-}
-
-func (wh *gatewayValidationWebhook) validateList(ctx context.Context, rawJson []byte, dryRun bool) (*validation.Reports, *multierror.Error) {
-	var (
-		ul      unstructured.UnstructuredList
-		reports *validation.Reports
-		errs    *multierror.Error
-	)
-
-	if err := ul.UnmarshalJSON(rawJson); err != nil {
-		return nil, &multierror.Error{Errors: []error{WrappedUnmarshalErr(err)}}
-	}
-	if reports, errs = wh.validator.ValidateList(ctx, &ul, dryRun); errs != nil {
-		return reports, errs
+	if isDelete {
+		return wh.deleteRef(ctx, gvk, ref, admissionRequest)
+	} else {
+		return wh.validateGvk(ctx, gvk, ref, admissionRequest)
 	}
```

Comment:
nit: its more go like not to nest this final else
```suggestion
	} 
	return wh.validateGvk(ctx, gvk, ref, admissionRequest)
	
```